tracing/tracingtest/oteltesttracer.go

```diff
@@ -0,0 +1,266 @@
+/*
+Package tracingtest provides an OpenTelemetry implementation for testing purposes.
+*/
+package tracingtest
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/propagation"
+	sdk "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	otelembedded "go.opentelemetry.io/otel/trace/embedded"
+)
+
+// OtelTracer is an implementation of opentelemetry.OtelTracer for testing. It records
+// the defined spans during a series of operations.
+type OtelTracer struct {
```

Comment:
I am wondering  if we should take a chance and get rid of tracing and tracing/tracingtest packages and just use otel directly (and https://pkg.go.dev/go.opentelemetry.io/otel/oteltest).

Reply:
In any case why do we implement own test tracer instead of using https://pkg.go.dev/go.opentelemetry.io/otel/oteltest ?

Reply:
Good catch. This is a mistake, I didn't know oteltest package, this seems to be a better solution for tracing tests for now on. But maybe we could address this in a different PR? It seems to me this will bring a lot of new changes to an already very big PR. WDYT?